src/components/api.ts

```diff
@@ -125,7 +142,7 @@ export default class Api {
           this.transport.setEventLoopReadSpeed(1);
           if (msgPacket.payload.length !== 4) {
             clearListeners();
-            reject('Data lenght is not 4, unable to proceed!');
+            reject(new Error('Data lenght is not 4, unable to proceed!'));
```

Comment:
You want to clear the timeout here as well, otherwise it will trigger after the rejection.

Reply:
Good catch! Will add the clearTimeout.

Reply:
ti is *length*